LeaderboardBackend/Services/Impl/AccountConfirmationService.cs

```diff
@@ -0,0 +1,82 @@
+using LeaderboardBackend.Models.Entities;
+using Microsoft.Extensions.Options;
+using NodaTime;
+
+namespace LeaderboardBackend.Services;
+
+public class AccountConfirmationService : IAccountConfirmationService
+{
+    private readonly ApplicationContext _applicationContext;
+    private readonly IEmailSender _emailSender;
+    private readonly IClock _clock;
+    private readonly AppConfig _appConfig;
+
+    public AccountConfirmationService(
+        ApplicationContext applicationContext,
+        IEmailSender emailSender,
+        IClock clock,
+        IOptions<AppConfig> appConfig
+    )
+    {
+        _applicationContext = applicationContext;
+        _emailSender = emailSender;
+        _clock = clock;
+        _appConfig = appConfig.Value;
+    }
+
+    public async Task<AccountConfirmation?> GetConfirmationById(Guid id)
+    {
+        return await _applicationContext.AccountConfirmations.FindAsync(id);
+    }
+
+    public async Task<CreateConfirmationResult> CreateConfirmationAndSendEmail(User user)
+    {
+        if (user.Role is not UserRole.Registered)
+        {
+            return new BadRole();
+        }
+
+        Instant now = _clock.GetCurrentInstant();
+
+        AccountConfirmation newConfirmation =
+            new()
+            {
+                CreatedAt = now,
+                ExpiresAt = now + Duration.FromHours(1),
+                UserId = user.Id,
+            };
+
+        _applicationContext.AccountConfirmations.Add(newConfirmation);
+
+        await _applicationContext.SaveChangesAsync();
+
+        try
+        {
+            await _emailSender.EnqueueEmailAsync(
+                user.Email,
+                "Confirm Your Account",
+                GenerateAccountConfirmationEmailBody(user, newConfirmation)
+            );
+        }
+        catch
+        {
+            // TODO: Log/otherwise handle the fact that the email failed to be queued - zysim
+            return new EmailFailed();
```

Comment:
What's wrong with using the default logging capabilities of ASP.NET? https://learn.microsoft.com/en-us/aspnet/core/fundamentals/logging

Reply:
Which library should we use? I hesitated on using Console because I thought we wouldn't want it

Reply:
You could try EventSource.

Reply:
Injected `ILogger<ClassName>` is the way.

But as I said in the email discord forum post, I don't even think this case should be handled at all.